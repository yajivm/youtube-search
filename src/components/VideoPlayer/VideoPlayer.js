import React from 'react';
import PropTypes from 'prop-types';
import { getPublishedDate } from '../../helpers/utils';
import Config from '../../config';

import './VideoPlayer.style.css';

const VideoPlayer = ({ videoId, videoData }) => {
    if (!videoId) {
        return (
            <div className="place-holder-text-wrapper" data-testid="place-holder-text-wrapper">
                Search for a video
            </div>
        );
    }

    const getVideoUrl = () => {
        return `${Config.EMBED_VIDEO_URL}${videoId}`;
    };

    const getVideoChannelDetails = () => {
        if (videoData && Object.keys(videoData).length > 0) {
            return (
                <ul className="details-list">
                    <li data-testid="channel-title">{videoData?.channelTitle}</li>
                    <li data-testid="published-date">{getPublishedDate(videoData?.publishedAt)}</li>
                </ul>
            )
        }
    };

    return (
        <div className="video-player" data-testid="video-player">
            <iframe
                title={videoId}
                className="video-iframe"
                data-testid="video-iframe"
                src={getVideoUrl()}
            />
            {videoData?.title && <h3 className="title" data-testid="video-title">{videoData?.title}</h3>}
            {getVideoChannelDetails()}
        </div>
    );
};

VideoPlayer.propTypes = {
    videoId: PropTypes.string,
    videoData: PropTypes.shape({}),
};

VideoPlayer.defaultProps = {
    videoId: null,
    title: null,
    channelDetails: {}
};

export default VideoPlayer;