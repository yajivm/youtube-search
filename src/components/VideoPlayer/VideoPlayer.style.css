.video-player {
  height: 600px;
}

.video-iframe {
  width: 100%;
  height: 100%;
}

.place-holder-text-wrapper {
  text-align: center;
  font-size: 18px;
  font-weight: bold;
}

.details-list {
    list-style: none;
    font-size: 12px;
    margin: 0;
    padding: 0;
}

.details-list > li {
  display: inline;
}

.details-list > li:first-child::after {
    content: "â€¢";
    margin: 0 4px;
}