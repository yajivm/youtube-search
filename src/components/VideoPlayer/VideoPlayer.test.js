import React from 'react';
import { render } from '@testing-library/react';
import Config from '../../config';

import VideoPlayer from './VideoPlayer';

const mockGetPublishedData = jest.fn();
jest.mock('../../helpers/utils', () => {
    return {
        getPublishedDate: args => mockGetPublishedData(args),
    }
});

let mockVideoId = "";
const mockVideoData = {
    channelTitle: '',
    publishedAt: '',
    title: ''
};

test('should render place holder text when video id is not valid', () => {
    const { getByTestId } = render(<VideoPlayer videoId={mockVideoId} videoData={mockVideoData} />);
    
    const placeHolderText = getByTestId('place-holder-text-wrapper');

    expect(placeHolderText).toBeDefined();
    expect(placeHolderText.textContent).toStrictEqual('Search for a video');
    expect(placeHolderText).toMatchSnapshot();
});

test('should render video player iframe when video id is valid', () => {
    mockVideoId = 'aTestId';

    const { getByTestId } = render(<VideoPlayer videoId={mockVideoId} videoData={mockVideoData} />);
    
    expect(getByTestId('video-player')).toBeDefined();
    expect(getByTestId('video-iframe').src).toStrictEqual(`${Config.EMBED_VIDEO_URL}aTestId`)
});

test('should render video title when video title prop is valid', () => {
    mockVideoData.title = 'aTestTitle';

    const { getByTestId } = render(<VideoPlayer videoId={mockVideoId} videoData={mockVideoData} />);
    
    expect(getByTestId('video-title')).toBeDefined();
    expect(getByTestId('video-title').textContent).toStrictEqual('aTestTitle');
});

test('should render channel details and date when publishedAt and channelTitle props are valid', () => {
    jest.resetAllMocks();
    mockVideoData.channelTitle = 'aChannelTestTitle';
    mockVideoData.publishedAt = '2021-11-12T22:18:07Z';

    const { getByTestId } = render(<VideoPlayer videoId={mockVideoId} videoData={mockVideoData} />);
    
    expect(getByTestId('channel-title')).toBeDefined();
    expect(getByTestId('channel-title').textContent).toStrictEqual('aChannelTestTitle');
    expect(mockGetPublishedData).toHaveBeenCalledTimes(1);
    expect(mockGetPublishedData).toHaveBeenCalledWith('2021-11-12T22:18:07Z');
    expect(getByTestId('published-date')).toBeDefined();
});