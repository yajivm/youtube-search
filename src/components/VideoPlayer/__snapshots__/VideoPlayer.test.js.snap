// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render place holder text when video id is not valid 1`] = `
<div
  class="place-holder-text-wrapper"
  data-testid="place-holder-text-wrapper"
>
  Search for a video
</div>
`;
